app/code/Magento/Braintree/view/frontend/web/js/view/payment/method-renderer/hosted-fields.js

```diff
@@ -148,6 +149,7 @@ define([
          */
         placeOrderClick: function () {
             if (this.validateCardType()) {
+                fullScreenLoader.startLoader();
```

Comment:
I think it's a better solution to disable the button after clicking instead showing loader. The possible fix - disabling place order action: `this.isPlaceOrderActionAllowed(false);` instead `fullScreenLoader.startLoader();`.

Reply:
Hi @joni-jones 

I appreciate your view on that and it completely aligned with my view 1.5 year ago, however, deactivating button still give a client todo some of the actions on page which might affect payment process (we have had such situations).

A few examples on "actions":
- Click somewhere on a page and got redirected somewhere from checkout page (as result the client will be charged, but no order stored in admin)
- Add / cancel coupon
- Add / cancel gift card
- Any other actions on a page

Ideally, once Order button is triggered, the client should be restricted doing anything on the page.

Regards,
Maxim

Reply:
I agree with you but when a customer clicks on `Place Order` button the place order action will be triggered, so loader appears even without your fix. Prosed fix prevents "doubleclicking", your fix masks a problem.